### PR TITLE
feat: Add Gen QA w/ Prompt Injection Detection for English

### PR DESCRIPTION
**Pull request by sjrl**
- [x] Need to test in dC

### CODE REVIEW

File: pipelines/GenerativeQuestionAnswering_prompt_injection.yaml [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+      similarity: cosine
+  - name: QueryClassifier
+    params:
+      model_name_or_path: deepset/deberta-v3-base-injection
+      labels: ['LEGIT','INJECTION']
+    type: TransformersQueryClassifier
+  - name: ErrorNode
+    type: ReturnError
+    params:
+      error_message: "Prompt injection detected so no answer is provided." # Error Message will be in metadata and no answer returned
```
Comment (sjrl) [inline pipelines/GenerativeQuestionAnswering_prompt_injection.yaml:26]: Hey @mkkuemmel to keep the language consistent between the German and English template should this be updated to "Prompt injection detected. No response is generated."?

File: pipelines/GenerativeQuestionAnswering_prompt_injection.yaml [hunk 2]
```diff
@@ -0,0 +1,129 @@
+# If you need help with the YAML format, have a look at https://docs.cloud.deepset.ai/docs/create-a-pipeline#create-a-pipeline-using-yaml.
+# This is a friendly editor that helps you create your pipelines with autosuggestions. To use them, press control + space on your keyboard.
+# Whenever you need to specify a model, this editor helps you out as well. Just type your Hugging Face organization and a forward slash (/) to see available models.
+
+# A RAG pipeline that generates answers using OpenAI’s GPT-3.5 model with a question answering prompt.
+# It classifies incoming queries and raises an error for prompt injection attempts. It’s optimized for English.
```
Comment (agnieszka-m) [inline pipelines/GenerativeQuestionAnswering_prompt_injection.yaml:6]: We should actually say here: It classifies incoming queries into genuine queries and prompt injection attempts. For prompt injection attempts, it adds an error to the metadata (whose metadata?) and returns "no answer".

File: pipelines/GenerativeQuestionAnswering_prompt_injection.yaml (hunk 1 above)
Comment (agnieszka-m) [reply pipelines/GenerativeQuestionAnswering_prompt_injection.yaml:26]: if we update the description, the comment here should be: The error message added to the metadata.

File: pipelines/GenerativeQuestionAnswering_prompt_injection.yaml [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    type: DeepsetCloudDocumentStore
+    params:
+      embedding_dim: 768
+      similarity: cosine
+  - name: QueryClassifier
+    params:
+      model_name_or_path: deepset/deberta-v3-base-injection
+      labels: ['LEGIT','INJECTION']
+    type: TransformersQueryClassifier
+  - name: ErrorNode
```
Comment (agnieszka-m) [inline pipelines/GenerativeQuestionAnswering_prompt_injection.yaml:23]: yet another custom node? :) 

let's add a comment: A node that adds an error message to the (whose?) metadata if a prompt injection is detected.

File: pipelines/GenerativeQuestionAnswering_prompt_injection.yaml [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+# Type is the class name of the component. 
+components:
+  - name: DocumentStore
+    type: DeepsetCloudDocumentStore
+    params:
+      embedding_dim: 768
+      similarity: cosine
+  - name: QueryClassifier
+    params:
+      model_name_or_path: deepset/deberta-v3-base-injection
```
Comment (agnieszka-m) [inline pipelines/GenerativeQuestionAnswering_prompt_injection.yaml:20]: is this a model we trained for detecting prompt injection? should we add a comment here?
"A model we trained to classify queries into prompt injection attempts and genuine queries."

File: pipelines/GenerativeQuestionAnswering_prompt_injection.yaml (hunk 2 above)
Comment (sjrl) [reply pipelines/GenerativeQuestionAnswering_prompt_injection.yaml:6]: The `answers` metadata

File: pipelines/GenerativeQuestionAnswering_prompt_injection.yaml (hunk 3 above)
Comment (sjrl) [reply pipelines/GenerativeQuestionAnswering_prompt_injection.yaml:23]: The `answers` meta data.

File: pipelines/GenerativeQuestionAnswering_prompt_injection.yaml (hunk 4 above)
Comment (sjrl) [reply pipelines/GenerativeQuestionAnswering_prompt_injection.yaml:20]: Yes it's a model that Jasper trained and that is a good description.

File: pipelines/GenerativeQuestionAnswering_prompt_injection.yaml (hunk 1 above)
Comment (mkkuemmel) [reply pipelines/GenerativeQuestionAnswering_prompt_injection.yaml:26]: Yes, we could align it with the German version, but it's ok to leave it this way as well in my opinion!

**Review by agnieszka-m — APPROVED** [review]

**Pull request merged**